server/services/case-service.js

```diff
@@ -31,25 +31,30 @@ const createCaseService = apiUrl => {
       const res = await request(`${apiUrl}/cases/${caseId}`)
       return res.data
     },
-    getMatchDetails: async defendantId => {
-      const res = await request(
+    getMatchDetails: async (defendantId, showAllMatches) => {
+      showAllMatches = Boolean(showAllMatches)
+      const res = (await request(
         `${apiUrl}/defendant/${defendantId}/matchesDetail`
-      ) || { data: {} }
+      ) || { data: {} })
 
       if (settings.enableMatcherLogging) {
         trackEvent('PiCMatcherLoggingServiceFirst', res)
       }
       // Sort offenderMatchDetails based on matchProbability in descending order
       if (Array.isArray(res.data.offenderMatchDetails)) {
         res.data.offenderMatchDetails.sort((a, b) => {
-          const probA = !isNaN(parseFloat(a.matchProbability))
-            ? parseFloat(a.matchProbability)
-            : 0
-          const probB = !isNaN(parseFloat(b.matchProbability))
-            ? parseFloat(b.matchProbability)
-            : 0
+          const probA = parseFloat(a.matchProbability ?? 0)
+          const probB = parseFloat(b.matchProbability ?? 0)
```

Comment:
is this just reverting the previous PR? I am confused by this

Reply:
yes, to more extent it is reverting the previous change